crates/polars-core/src/hashing/vector_hasher.rs

```diff
@@ -283,24 +300,83 @@ impl VecHash for BooleanChunked {
             }
             offset += arr.len();
         });
+        Ok(())
     }
 }
 
 impl VecHash for Float32Chunked {
-    fn vec_hash(&self, random_state: RandomState, buf: &mut Vec<u64>) {
-        self.bit_repr_small().vec_hash(random_state, buf)
+    fn vec_hash(&self, random_state: RandomState, buf: &mut Vec<u64>) -> PolarsResult<()> {
+        self.bit_repr_small().vec_hash(random_state, buf)?;
+        Ok(())
     }
 
-    fn vec_hash_combine(&self, random_state: RandomState, hashes: &mut [u64]) {
-        self.bit_repr_small().vec_hash_combine(random_state, hashes)
+    fn vec_hash_combine(&self, random_state: RandomState, hashes: &mut [u64]) -> PolarsResult<()> {
+        self.bit_repr_small()
+            .vec_hash_combine(random_state, hashes)?;
+        Ok(())
     }
 }
 impl VecHash for Float64Chunked {
-    fn vec_hash(&self, random_state: RandomState, buf: &mut Vec<u64>) {
-        self.bit_repr_large().vec_hash(random_state, buf)
+    fn vec_hash(&self, random_state: RandomState, buf: &mut Vec<u64>) -> PolarsResult<()> {
+        self.bit_repr_large().vec_hash(random_state, buf)?;
+        Ok(())
     }
-    fn vec_hash_combine(&self, random_state: RandomState, hashes: &mut [u64]) {
-        self.bit_repr_large().vec_hash_combine(random_state, hashes)
+    fn vec_hash_combine(&self, random_state: RandomState, hashes: &mut [u64]) -> PolarsResult<()> {
+        self.bit_repr_large()
+            .vec_hash_combine(random_state, hashes)?;
+        Ok(())
+    }
+}
+
+#[cfg(feature = "groupby_list")]
+impl VecHash for ListChunked {
+    fn vec_hash(&self, _random_state: RandomState, _buf: &mut Vec<u64>) -> PolarsResult<()> {
+        polars_ensure!(
+        self.inner_dtype().to_physical().is_numeric(),
+        ComputeError: "grouping on list type is only allowed if the inner type is numeric"
+        );
+        _buf.clear();
+        _buf.reserve(self.len());
+        let null_h = get_null_hash_value(_random_state.clone());
+
+        for arr in self.downcast_iter() {
+            _buf.extend(
+                numeric_list_bytes_iter(arr)?.map(|opt_bytes| match opt_bytes {
+                    Some(s) => _random_state.hash_one(s),
```

Comment:
I'm a bit unsure whether using `hash_one` or `xxh3_64_with_seed` is more appropriate here, or if some usage is incorrect.

Reply:
I think we should go with `xxh3_64_with_seed`. I believe that one is faster.